torchgeo/datasets/inaturalist.py

```diff
@@ -44,9 +44,9 @@ def __init__(self, root: str = "data") -> None:
         """
         super().__init__()
 
-        self.root = root
+        self.root = str(root)
 
-        files = glob.glob(os.path.join(root, "**.csv"))
+        files = glob.glob(os.path.join(self.root, "**.csv"))
```

Comment:
This doesn't need to be a str, right?

Reply:
This will need to be ``str`` as ``join`` does not support concatenation of ``bytes`` and ``str``.